### PR TITLE
Adds a temporary workaround for a BYOND bug

### LINKED ISSUE #32305: [HIGH PRIORITY] Machines (with components) do not work at all.
Confirmed to be a tgstation issue.

Related issues:
https://github.com/HippieStation/HippieStation/issues/4468
https://github.com/HippieStation/HippieStation/issues/4469
https://github.com/HippieStation/HippieStation/issues/4463

Tested locally.

Occurred in some pr after https://github.com/tgstation/tgstation/commit/fc92519d451935ecca345d7ffac1ec83e978e936

### PR DESCRIPTION
**Pull request by Cyberboss**
This broke components

Fixes #32305

### CODE REVIEW

**Comment by duncathan** [comment]
my question is why the fuck this hack works

**Comment by optimumtact** [comment]
because of this http://www.byond.com/forum/?post=73469#comment23285352

File: code/datums/components/_component.dm [hunk 1]
```diff
@@ -137,6 +137,10 @@
 	return
 
 /datum/component/proc/_GetInverseTypeList(our_type = type)
+#if DM_VERSION > 511
+#warning Remove this hack for http://www.byond.com/forum/?post=73469
```
Comment (RemieRichards) [inline code/datums/components/_component.dm]: or just:
```
if(!our_type)
    our_type = type
```
No need for nonsense (and, imo, an ugly use of default arguments)

Comment (RemieRichards) [reply code/datums/components/_component.dm]: Especially since, as lummox reveals in the bug report, default args just generate the if anyway!

Comment (Cyberboss) [reply code/datums/components/_component.dm]: The bug is actually the var/current_type = parent_type line below this. src is null because of some fuckery

Comment (RemieRichards) [reply code/datums/components/_component.dm]: and its caused BY the default arg.

Comment (duncathan) [reply code/datums/components/_component.dm]: yeah this workaround was discussed in the thread and is far more elegant than what's been done here

**Pull request merged**